pkg/mimirtool/client/client.go

```diff
@@ -119,12 +122,26 @@ func (r *MimirClient) doRequest(path, method string, payload []byte) (*http.Resp
 		return nil, err
 	}
 
+	if (r.user != "" || r.key != "") && r.authToken != "" {
+		err := errors.New("at most one of basic auth or auth token should be configured")
+		log.WithFields(log.Fields{
+			"url":    req.URL.String(),
+			"method": req.Method,
+			"error":  err,
+		}).Errorln("error during request to mimir api")
+		return nil, err
+	}
+
 	if r.user != "" {
 		req.SetBasicAuth(r.user, r.key)
 	} else if r.key != "" {
 		req.SetBasicAuth(r.id, r.key)
 	}
 
+	if r.authToken != "" {
```

Comment:
Should this be part of the block above as another `else if` case since we don't want it set when basic auth is being used?

Reply:
Thanks, Addressed.